sdk/src/transaction.rs

```diff
@@ -330,6 +334,27 @@ impl Transaction {
         }
     }
 
+    pub fn verify_secp256k1(&self) -> Result<()> {
+        for instruction in &self.message().instructions {
+            if instruction.program_id_index as usize >= self.message().account_keys.len() {
```

Comment:
This isn't verified in another common place?

Reply:
Yes, but it's quite a bit later than this operation in the pipeline. I can move it to before this verify is called, but I also need to then have it in two different locations because this part is not shared with the replay part. That refactoring would also potentially change the error behavior for non-secp things.

Reply:
Can we add a comment here that the TX may not be sanitized yet and that care should be taken with indices?